aten/src/ATen/native/transformers/cuda/flash_attn/fmha_api.cpp

```diff
@@ -215,31 +295,213 @@ mha_fwd(const at::Tensor &q,         // total_q x num_heads x head_size, total_q
                      max_seqlen_k,
                      num_heads,
                      head_size,
-                     q, k, v,
+                     q, k, v, out,
                      cu_seqlens_q.data_ptr(),
                      cu_seqlens_k.data_ptr(),
-                     o.data_ptr(),
                      loop ? o_tmp.data_ptr() : nullptr,
                      nullptr,
                      softmax_lse.data_ptr(),
                      p_dropout,
                      softmax_scale,
-                     is_causal);
+                     is_causal,
+                     num_splits);
 
-    run_fmha_fprop(launch_params, /*configure=*/ true);
     // number of times random will be generated per thread, to offset philox counter in thc random
     // state
-    int64_t counter_offset = launch_params.elts_per_thread;
+    // We use a custom RNG that increases the offset by batch_size * nheads * 32.
+    int64_t counter_offset = launch_params.params.b * launch_params.params.h * 32;
+    at::PhiloxCudaState rng_engine_inputs;
```

Comment:
This line is breaking a number of internal jobs, since they are building PyTorch with the `-Wunused-variable` flag. See the jobs and logs here: [D42733435](https://www.internalfb.com/diff/D42733435)

Small snippet for reference:
```
[2023-01-24T20:32:00.241-08:00] Stderr: fbcode/caffe2/aten/src/ATen/native/transformers/cuda/flash_attn/fmha_api.cpp:313:25: error: unused variable 'rng_engine_inputs' [-Werror,-Wunused-variable]
[CONTEXT] [2023-01-24T20:32:00.241-08:00]     at::PhiloxCudaState rng_engine_inputs;
[CONTEXT] [2023-01-24T20:32:00.241-08:00]                         ^
[2023-01-24T21:09:33.507-08:00] Stderr: fbcode/caffe2/aten/src/ATen/native/transformers/cuda/flash_attn/fmha_api.cpp:313:25: error: unused variable 'rng_engine_inputs' [-Werror,-Wunused-variable]
[CONTEXT] [2023-01-24T21:09:33.507-08:00]     at::PhiloxCudaState rng_engine_inputs;
[CONTEXT] [2023-01-24T21:09:33.507-08:00]   
```

Reply:
Hmm, we have a bunch of `-Wno-unused-but-set-variable` etc in CMake, perhaps it affected the output. 
Let's create a followup issue to investigate, why wasn't it caught

Reply:
cc @drisspg

Reply:
I've [removed](https://github.com/HazyResearch/flash-attention/commit/6b4a48218edb55fb67e087f4df8d7ba4711e75bb) this unused variable in FlashAttention source code.

Reply:
Removed from core version here: #93024